src/useEntity.js

```diff
@@ -29,7 +29,9 @@ export const useEntity = (
     return () => {
       for (let i = 0, c = entity._subscribers.length; i < c; i++) {
         if (entity._subscribers[i] === subscriberFn) {
-          entity._subscribers[i] = null
+          // Move the last subscriber over the removed one and pop it 
+          entity._subscribers[i] = entity._subscribers[entity._subscribers.length - 1]
+          entity._subscribers.pop()
```

Comment:
Alternatively just replace the whole callback with
```js
entity._subscribers = entity._subscribers.filter(fn => fn !== subscriberFn)
```

Reply:
^ This is preferable if the order of `_subscibers` should not change

Reply:
Please check `entity.js`, in `createSetter` function.  There is an existing cleanup there, which was intentionally deferred to make sure the subscriber callback loop is complete before it removes items from the array.

Do you think moving that to the `useEffect` cleanup will be better? If I remember correctly, in my prior art (react-entities), it was originally in the effect cleanup, but there was a timing issue that was reported, so I moved it to the current location. This new library is based on that same core, so I figured the fix carried over here.

What do you think?

Reply:
Thanks for this. Please refer to my general comment on this PR.